azure_batch/main.tf

```diff
@@ -0,0 +1,183 @@
+      }
+      snapshot {
+        delete_after_days_since_creation_greater_than = var.keep_inp_files_during_n_days
+      }
+    }
+  }
+}
+
+# Create a container registry
+resource "azurerm_container_registry" "acr" {
+  name                = "batchComputingContainerRegistry"
+  resource_group_name = var.resource_group_name
+  location            = var.resource_group_location
+  sku                 = "Basic"
+  admin_enabled       = true
+}
+
+# Push a Docker image to the registry
+resource "null_resource" "push_to_acr" {
+  provisioner "local-exec" {
+    command = <<EOF
+          az acr login --name "${azurerm_container_registry.acr.login_server}" --username "${azurerm_container_registry.acr.admin_username}" --password "${azurerm_container_registry.acr.admin_password}"
+          docker tag ${var.image_name}:${var.image_tag} "${azurerm_container_registry.acr.login_server}"/${var.image_name}:${var.image_tag}
+          docker push "${azurerm_container_registry.acr.login_server}"/${var.image_name}:${var.image_tag}
+  EOF
+  }
+}
+
+
+# Create an Azure Batch account
+resource "azurerm_batch_account" "batch_account" {
+  name                 = "batchaccount${terraform.workspace}"
+  resource_group_name  = var.resource_group_name
+  location             = var.resource_group_location
+  pool_allocation_mode = "BatchService"
+  storage_account_id   = azurerm_storage_account.batch_computing_s_a.id
+}
+
+# Create the pool within the Azure Batch account
+resource "azurerm_batch_pool" "dynamic_pool" {
+  name                = "pool-${terraform.workspace}"
+  resource_group_name = var.resource_group_name
+  account_name        = azurerm_batch_account.batch_account.name
+  display_name        = "Time-based autoscaled pool"
+  vm_size             = var.vm_size
+  node_agent_sku_id   = "batch.node.ubuntu 20.04"
+
+  auto_scale {
+    evaluation_interval = "PT5M"
+
+    # Autoscale formula (see: https://docs.microsoft.com/en-us/azure/batch/batch-automatic-scaling)
+    formula = var.autoscale_formula
+  }
+
+  # Warning: only the 'microsoft-azure-batch' publisher provides Docker-compatible images
+  storage_image_reference {
+    publisher = "microsoft-azure-batch"
+    offer     = "ubuntu-server-container"
+    sku       = "20-04-lts"
+    version   = "latest"
+  }
+
+  container_configuration {
+    type = "DockerCompatible"
+    container_registries {
+      # Put informations about the private container registry here
+      registry_server = azurerm_container_registry.acr.login_server
+      user_name       = azurerm_container_registry.acr.admin_username
+      password        = azurerm_container_registry.acr.admin_password
+    }
+
+    # Pull the image stored in the container registry
+    container_image_names = ["${azurerm_container_registry.acr.login_server}/${var.image_name}:${var.image_tag}"]
+  }
+
+  start_task {
+    # Replace the following by the Docker entrypoint
```

Comment:
What do you mean with this comment?

Reply:
I just removed the `start_task` part which doesn't really makes sense if we use a Docker image.